base/settings/tests/testthat/test.addSecrets.R

```diff
@@ -0,0 +1,65 @@
+test_that("`addSecrets` returns settings when `~/.pecan.xml` does not exist", {
+  settings <- list()
+  expect_equal(addSecrets(settings), settings)
+})
+
+test_that("`addSecrets` returns settings when force is FALSE and secrets have already been added", {
+  settings <- list(
+    settings.info = list(
+      secrets.added = TRUE
+    )
+  )
+  mockery::stub(addSecrets, 'file.exists', TRUE)
+  expect_equal(addSecrets(settings, force = FALSE), settings)
+})
+
+test_that("`addSecrets` adds secret settings when force is TRUE and secrets have already been added", {
+  settings <- list(
+    settings.info = list(
+      secrets.added = TRUE
+    )
+  )
+
+  m <- list(
```

Comment:
A more informative name would be helpful here -- am I right this is something like the `mock_xmlToList_result`?